src/internal/group_api/mod.rs

```diff
@@ -359,88 +372,153 @@ pub(crate) fn get_group_keys<'a>(
-            if member_ids_and_keys.len() != members.len() {
-                // figure out which user ids could not be found in the database and include them in the error message.
-                use std::{collections::HashSet, iter::FromIterator};
-                let desired_members_set: HashSet<&UserId> = HashSet::from_iter(members);
-                let found_members: Vec<UserId> =
-                    member_ids_and_keys.into_iter().map(|(x, _)| x).collect();
-                let found_members_set: HashSet<&UserId> = HashSet::from_iter(&found_members);
-                let diff: HashSet<&&UserId> =
-                    desired_members_set.difference(&found_members_set).collect();
-                futures::future::err(IronOxideErr::UserDoesNotExist(format!(
-                    "Failed to find the following users from the `members` list: {:?}",
-                    diff
-                )))
-            } else {
-                transform::gen_group_keys(recrypt)
-                    .and_then(move |(plaintext, group_priv_key, group_pub_key)| {
-                        // encrypt the group secret to the current user as they will be the group admin
-                        let encrypted_group_key = recrypt.encrypt(
-                            &plaintext,
-                            &user_master_pub_key.into(),
-                            &auth.signing_private_key().into(),
-                        )?;
-                        // Map from UserId to (PublicKey, Optional TransformKey)
-                        // TransformKey is optional because we need it for members, but won't need it
-                        // when we expand this to include admins
-                        let member_info_result: Result<
-                            HashMap<UserId, (PublicKey, Option<TransformKey>)>,
-                            _,
-                        > = member_ids_and_keys
-                            .into_iter()
-                            .map(|(id, user_pub_key)| {
-                                let maybe_transform_key = recrypt.generate_transform_key(
-                                    &group_priv_key.clone().into(),
-                                    &user_pub_key.clone().into(),
-                                    &auth.signing_private_key().into(),
-                                );
-                                maybe_transform_key.map(|transform_key| {
-                                    (id, (user_pub_key, Some(transform_key.into())))
-                                })
+    use futures::future::Either;
+    let owner_id = match owner.clone() {
+        Some(id) => id,
+        None => auth.account_id().clone(),
+    };
+    // don't make any requests if validation fails
+    if admins.is_empty() {
+        Either::A(err(IronOxideErr::ValidationError(
+            format!("admins"),
+            format!("admins list cannot be empty"),
+        )))
+    } else if !admins.contains(&owner_id) {
```

Comment:
You can just check if add_as_admin is false and owner = none

Reply:
This would miss the case where the caller adds themselves to the `admins` list manually, which we decided takes precedence over `add_as_admin`. But I realized that my `if` and `else if` really only covered one thing together, so I cleaned them up as a single check.